src/main/java/uk/gov/register/store/postgres/PostgresDriverTransactional.java

```diff
@@ -31,6 +29,9 @@ private PostgresDriverTransactional(Handle handle, MemoizationStore memoizationS
         this.stagedEntries = new ArrayList<>();
         this.stagedItems = new LinkedHashSet<>();
         this.stagedCurrentKeys = new HashMap<>();
+
+        // Initialise the entry cutoff now that have have our PostgresDriver created
+        ((TransactionalMemoizationStore)memoizationStore).setCurrentEntryCount(this.getTotalEntries());
```

Comment:
we can pass in `TransactionalMemoizationStore` in private constructor and remove this casting, what do you think?